src/tools/tidy/src/style.rs

```diff
@@ -252,10 +283,10 @@ pub fn check(path: &Path, bad: &mut bool) {
         // We only check CSS files in rustdoc.
         path.extension().map_or(false, |e| e == "css") && !is_in(path, "src", "librustdoc")
     }
-
-    let problematic_consts_strings: Vec<String> = (PROBLEMATIC_CONSTS.iter().map(u32::to_string))
-        .chain(PROBLEMATIC_CONSTS.iter().map(|v| format!("{:x}", v)))
-        .chain(PROBLEMATIC_CONSTS.iter().map(|v| format!("{:X}", v)))
+    let problematic_consts = generate_problems(ROOT_PROBLEMATIC_CONSTS, &[('A', '4'), ('B', '8'), ('E', '3')].iter().cloned().collect()).collect::<Vec<u32>>();
+    let problematic_consts_strings: Vec<String> = (problematic_consts.iter().map(u32::to_string))
+        .chain(problematic_consts.iter().map(|v| format!("{:x}", v)))
+        .chain(problematic_consts.iter().map(|v| format!("{:X}", v)))
         .collect();
     let problematic_regex = RegexSet::new(problematic_consts_strings.as_slice()).unwrap();
```

Comment:
Since the matching is done here using a RegexSet anyway, I wonder if leaning into the regex would be a much more straightforward and flexible way to implement this, instead of using that bit shift logic to try to enumerate every possible permutation of the digit choices.

Something like:

```rust
problematic_consts.iter().map(|int| {
    format!("{:x}", int)
        .replace("a", "[aA4]")
        .replace("b", "[bB8]")
        .replace("c", "[cC]")
        .replace("d", "[dD]")
        .replace("e", "[eE3]")
        .replace("f", "[fF]")
})
```

Reply:
Actually, why wouldn't it just work?